pennylane/ops/qubit.py

```diff
@@ -565,6 +567,176 @@ def decomposition(phi, theta, omega, wires):
         return decomp_ops
 
 
+class MultiRZ(Operation):
+    r"""MultiRZ(theta, wires)
+    Arbitrary multi Z rotation.
+
+    .. math::
+
+        MultiRZ(\theta) = \exp(-i \frac{\theta}{2} Z^{\otimes n})
+
+    **Details:**
+
+    * Number of wires: Any
+    * Number of parameters: 1
+    * Gradient recipe: :math:`\frac{d}{d\theta}f(MultiRZ(\theta)) = \frac{1}{2}\left[f(MultiRZ(\theta +\pi/2)) - f(MultiRZ(\theta-\pi/2))\right]`
+      where :math:`f` is an expectation value depending on :math:`MultiRZ(\theta)`.
+
+    .. note::
+
+        If the ``MultiRZ`` gate is not supported on the targeted device, PennyLane
+        will decompose the gate using :class:`~.RZ` and :class:`~.CNOT` gates.
+
+    Args:
+        theta (float): rotation angle :math:`\theta`
+        wires (Sequence[int] or int): the wires the operation acts on
+    """
+    num_params = 1
+    num_wires = Any
+    par_domain = "R"
+    grad_method = "A"
+
+    @staticmethod
+    def _matrix(*params):
```

Comment:
What if this was made an instance method and the `n=len(self.wires)` was used here? That would mean that `matrix` would not have to be defined (though transposing would be included from the inherited `matrix`)

Reply:
That is a good idea. I'm only a little bit hesitant to override a static method with a non-static one 🤔 But this must be improved in a way

Reply:
I left it as it is now. It is only used inside `MultiRZ` itself and in the tests. Keeping the static method allows one to get the matrix without having to instantiate a `MultiRZ` object which is useful. I added a docstring where this is noted.